### PR TITLE
[test] Adjust KVOKeyPaths test for back deployment

### PR DESCRIPTION
**Pull request by lorentey**
C.f. https://github.com/apple/swift/pull/40051

rdar://85274909

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci test

File: test/stdlib/KVOKeyPaths.swift [hunk 1]
```diff
@@ -216,5 +216,5 @@ testObjectForOptionalKeyPath.optionalObject = "foo"
 
 // CHECK-51-LABEL: observe keyPath with optional value
 // CHECK-51-NEXT: oldValue = Optional(nil), newValue = {{Optional\(nil\)|nil}}
```
Comment (mikeash) [inline test/stdlib/KVOKeyPaths.swift:218]: Might want to apply the same fix to this `Optional(nil)` while we're in here.

**Review by mikeash — APPROVED** [review]
One potential enhancement, but LGTM.

**Comment by lorentey** [comment]
```
19:23:09 Test Suite 'Selected tests' started at 2021-11-10 19:21:38.133
19:23:09 Test Suite 'SwiftPMPackageTests.xctest' started at 2021-11-10 19:21:38.134
19:23:09 Test Suite 'PackageCollectionsStorageTests' started at 2021-11-10 19:21:38.134
19:23:09 Test Case '-[PackageCollectionsTests.PackageCollectionsStorageTests testListNonBatching]' started.
19:23:09 
19:23:09 Exited with signal code 4
```

Huh.

File: test/stdlib/KVOKeyPaths.swift (hunk 1 above)
Comment (lorentey) [reply test/stdlib/KVOKeyPaths.swift:218]:
```suggestion
// CHECK-51-NEXT: oldValue = {{Optional\(nil\)|nil}}, newValue = {{Optional\(nil\)|nil}}
```

**Comment by lorentey** [comment]
@swift-ci test

**Pull request merged**